src/utils/CharactersPageUtils.js

```diff
@@ -83,21 +92,34 @@ export const prepareEditCharacterData = ({ formData, character }) => {
   } = formData
   const { id } = character
 
+  const characterSpecies = species
+    ? {
+        ...character.species,
+        id: species.id,
+        name: species.label,
+      }
+    : null
+
+  const characterHomeworld = homeworld
+    ? {
+        ...character.homeworld,
+        id: homeworld.id,
+        name: homeworld.label,
```

Comment:
Any data you want to keep from the original object should be read from the original object, not the form data